### PR TITLE
asyncloaddata: add table and helper functions 

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

### What is changed and how it works?

Add a table and helper function so that LOAD DATA worker can use them to run asynchronously

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/asyncloaddata/util.go [hunk 1]
```diff
@@ -0,0 +1,411 @@
+// Copyright 2023 PingCAP, Inc.
```
Comment (hawkingrei) [inline executor/asyncloaddata/util.go:1]: remember to run ```make bazel_prepare```.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
ping @D3Hunter

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**